### PR TITLE
Add SMS section to registration page

### PR DESCRIPTION
**Pull request by sam-warren**
## Status
**Ready**

## Fixes or Implements [AB#8334](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8334)
- [X] Enhancement
- [ ] Bug
- [ ] Documentation

## Description
Added SMS field to registration page and accompanying validation logic to ensure user info is entered correctly

### Steps to Reproduce:
If this is a bug, please provide details on how to reproduce the code.

### UI Changes
YES 

### Browsers Tested
- [X] Chrome
- [ ] Safari
- [ ] Edge
- [ ] Firefox

Screen Shot 2020-05-13 at 1 08 38 PM

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/app/views/registration.vue [hunk 1]
```diff
@@ -389,21 +465,31 @@ export default class RegistrationComponent extends Vue {
...
         }),
         sameAsEmail: sameAs("email"),
-        email
+        email,
+      },
+      preferredMethod: {
+        required
       },
-      accepted: { isChecked: sameAs(() => true) }
+      accepted: { isChecked: sameAs(() => true) },
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/app/views/registration.vue:492]: are these comas added by lintfix?

File: Apps/WebClient/src/ClientApp/app/views/registration.vue [hunk 2]
```diff
@@ -433,11 +519,11 @@ export default class RegistrationComponent extends Vue {
     this.loadingTermsOfService = true;
     this.userProfileService
       .getTermsOfService()
-      .then(result => {
+      .then((result) => {
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/app/views/registration.vue:522]: are these parenthesis added by lintfix?

Comment (sam-warren) [reply Apps/WebClient/src/ClientApp/app/views/registration.vue:522]: yep

File: Apps/WebClient/src/ClientApp/app/views/registration.vue (hunk 1 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/app/views/registration.vue:492]: weird, my lintfix removes them

File: Apps/WebClient/src/ClientApp/app/views/registration.vue (hunk 2 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/app/views/registration.vue:522]: same, mine removes them

**Pull request merged**